File: Day12_Isomorphic Strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int n1 = s.length();
        int n2 = t.length();
        
        if(n1 != n2)
        {
            return false;
        }
        
        int temp1[256];
        memset(temp1,-1,sizeof(temp1));
        int temp2[256];
        memset(temp2,-1,sizeof(temp2));
        
        for(int i=0;i<n1;i++)
        {
            if(temp1[s[i]] != temp2[t[i]])
            {
                return false;
            }
            
            temp1[s[i]]=i;
            temp2[t[i]]=i;
            
            
        }
        return true;
        
    
        
    }
};

File: Day12_Isomorphic Strings.py
class Solution:
    def isIsomorphic(self, str1: str, str2: str) -> bool:
        n1 = len(str1)
        n2 = len(str2)
        
        if n1!=n2:
            return False;
            
        temp1 = {}
        temp2 = {}
        
        for i in range(n1):
            if str1[i] in temp1:
                if temp1[str1[i]] != str2[i]:
                    return False
            else:
                temp1[str1[i]]=str2[i]
                    
            if str2[i] in temp2:
                if temp2[str2[i]] != str1[i]:
                    return False
                    
            else:
                temp2[str2[i]] = str1[i]
        
        return True
        
